tinygrad/tensor.py

```diff
@@ -821,8 +821,13 @@ def __itruediv__(self, x) -> Tensor: return self.assign(self.div(x))
   def __imatmul__(self, x) -> Tensor: return self.assign(self.matmul(x))
   def __ixor__(self, x) -> Tensor: return self.assign(self.xor(x))
 
-  def __lt__(self, x) -> Tensor: return mlops.Less.apply(*self._broadcasted(x, False))
-  def __gt__(self, x) -> Tensor: return mlops.Less.apply(*self._broadcasted(x, True))
+  # in webgpu bool cannot be used as a storage buffer type
+  def __lt__(self, x) -> Tensor:
+    out = mlops.Less.apply(*self._broadcasted(x, False))
+    return out.float() if self.device == "WEBGPU" else out
```

Comment:
can do this cast in mlops itself and no need to copy lines

Reply:
i think tinygrad should behave like torch
```
a = torch.tensor([1,2,3,4])
b = torch.tensor([1,2,3,4])
out = a < b
print(out.dtype) # bool
```

Reply:
i think tinygrad should behave like torch

Reply:
i agree `__lt__` should return dtypes.bool (same for `__le__`, `__eq__` etc), but the "WEBGPU" type conversion can happen in one place